examples/api/index.ts

```diff
@@ -42,16 +43,24 @@ const lambda = new aws.lambda.Function("myfunction", {
     runtime: aws.lambda.NodeJS8d10Runtime,
 });
 
-const api = new aws.apigateway.x.API("myapi", {
-    routes: [
-        { method: "GET", path: "/a", handler: async (event) => {
-            return {
-                statusCode: 200,
-                body: "<h1>Hello world!</h1>",
-            };
-        }},
-        { method: "GET", path: "/b", handler: lambda },
-    ],
+const api = new x.API("myapi", {
+    routes: {
+        "/a": {
+            kind: "EventHandler",
```

Comment:
So, i'm not a huge fan that users of the API have to specify this 'kind' themselves.  We could have a model where this was elided, and we used just the properties of the object to determine what sort of op this was.  However, we'd have to be super careful to make sure that each type of 'route' didn't overlap in terms of the properties they had.

@lukehoban what are your thoughts here?

Reply:
Yes - no kind.